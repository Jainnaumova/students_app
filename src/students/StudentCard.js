import React from 'react';

import './student-card.css';

export default props => {
  const { studentData } = props;
  return (
    <div>
      <div className='title'>Student Details</div>
      <div className='student-card'>
        <div>
          <div className='block'>
            <div className='sub-title'>Student ID</div>
            <div className='data'>{studentData.studentId}</div>
          </div>
          <div className='block'>
            <div className='sub-title'>Student Name</div>
            <div className='data'>{`${studentData.firstName} ${studentData.lastName}`}</div>
          </div>
          <div className='block'>
            <div className='sub-title'>School Name</div>
            <div className='data'>{studentData.schoolName}</div>
          </div>
          <div className='block'>
            <div className='sub-title'>School ID</div>
            <div className='data'>{studentData.schoolId}</div>
          </div>
          <div className='block'>
            <div className='sub-title'>District</div>
            <div className='data'>{studentData.district}</div>
          </div>
          <div className='block'>
            <div className='sub-title'>Email</div>
            <div className='data'>{studentData.email}</div>
          </div>
          <div className='block'>
            <div className='sub-title'>Grade</div>
            <div className='data'>{studentData.grade}</div>
          </div>
          <div className='block'>
            <div className='sub-title'>Cohort</div>
            <div className='data'>{studentData.cohort}</div>
          </div>
          <div className='block'>
            <div className='sub-title'>Attendance Percentage</div>
            <div className='data'>{studentData.attendancePercentage}</div>
          </div>
        </div>
        <div>
          <div className='block'>
            <div className='sub-title'>Guidance Counselor</div>
            <div className='data'>{studentData.guidanceCounselor}</div>
          </div>
          <div className='block'>
            <div className='sub-title'>Guidance Counselor Email</div>
            <div className='data'>{studentData.guidanceCounselorEmail}</div>
          </div>
          <div className='block'>
            <div className='sub-title'>Home Phone Number</div>
            <div className='data'>{studentData.homePhoneNumber}</div>
          </div>
          <div className='block'>
            <div className='sub-title'>Advisor</div>
            <div className='data'>{studentData.advisor}</div>
          </div>
          <div className='block'>
            <div className='sub-title'>Official Class</div>
            <div className='data'>{studentData.officialClass}</div>
          </div>
          <div className='block'>
            <div className='sub-title'>Total Credits Earned</div>
            <div className='data'>{studentData.totalCreditsEarned}</div>
          </div>
          <div className='block'>
            <div className='sub-title'>Has Passed Regents</div>
            <div className='data'>{studentData.hasPassedRegents ? 'YES' : 'NO'}</div>
          </div>
          <div className='block'>
            <div className='sub-title'>Status</div>
            <div className='data'>{studentData.status}</div>
          </div>
          <div className='block'>
            <div className='sub-title'>Admin Date</div>
            <div className='data'>{studentData.adminDate}</div>
          </div>
        </div>
      </div>
    </div>
  );
}
