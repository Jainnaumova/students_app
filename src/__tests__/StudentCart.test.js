import React from 'react';
import ReactDOM from 'react-dom';
import Adapter from 'enzyme-adapter-react-15';
import Enzyme, { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import toJson from 'enzyme-to-json';
import StudentCard from '../students/StudentCard';

Enzyme.configure({ adapter: new Adapter() });

let props = {
  studentData: {
    "firstName": "Adam",
    "lastName": "Tam",
    "studentId": "7395508",
    "schoolName": "New Visions HS",
    "schoolId": "123XYZ",
    "district": "1A",
    "email": "adam.tam@newvisionshs.org",
    "grade": 11,
    "cohort": 2020,
    "attendancePercentage": 78.07,
    "guidanceCounselor": "Kent, Clark",
    "guidanceCounselorEmail": "kent.clark@newvisionshs.org",
    "homePhoneNumber": "917-590-3079",
    "advisor": "Kyle, Selina",
    "officialClass": "345",
    "totalCreditsEarned": 24,
    "hasPassedRegents": true,
    "status": "Active",
    "adminDate": "Sept 2016"
  },
}

describe('StudentCard', () => {

  it('should match the shapshot', () => {
    const tree = renderer.create(
      <StudentCard { ...props }/>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should handle hasPassedRegents correctly', () => {
    const wrapperOne = shallow(<StudentCard {...props} />);
    expect(wrapperOne.find('div').at(52).text()).toEqual('YES');
    props.studentData.hasPassedRegents = false;
    const wrapperTwo = shallow(<StudentCard {...props} />);
    expect(wrapperTwo.find('div').at(52).text()).toEqual('NO');
  });

});
