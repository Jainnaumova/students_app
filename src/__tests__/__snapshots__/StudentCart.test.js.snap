// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StudentCard should match the shapshot 1`] = `
<div>
  <div
    className="title"
  >
    Student Details
  </div>
  <div
    className="student-card"
  >
    <div>
      <div
        className="block"
      >
        <div
          className="sub-title"
        >
          Student ID
        </div>
        <div
          className="data"
        >
          7395508
        </div>
      </div>
      <div
        className="block"
      >
        <div
          className="sub-title"
        >
          Student Name
        </div>
        <div
          className="data"
        >
          Adam Tam
        </div>
      </div>
      <div
        className="block"
      >
        <div
          className="sub-title"
        >
          School Name
        </div>
        <div
          className="data"
        >
          New Visions HS
        </div>
      </div>
      <div
        className="block"
      >
        <div
          className="sub-title"
        >
          School ID
        </div>
        <div
          className="data"
        >
          123XYZ
        </div>
      </div>
      <div
        className="block"
      >
        <div
          className="sub-title"
        >
          District
        </div>
        <div
          className="data"
        >
          1A
        </div>
      </div>
      <div
        className="block"
      >
        <div
          className="sub-title"
        >
          Email
        </div>
        <div
          className="data"
        >
          adam.tam@newvisionshs.org
        </div>
      </div>
      <div
        className="block"
      >
        <div
          className="sub-title"
        >
          Grade
        </div>
        <div
          className="data"
        >
          11
        </div>
      </div>
      <div
        className="block"
      >
        <div
          className="sub-title"
        >
          Cohort
        </div>
        <div
          className="data"
        >
          2020
        </div>
      </div>
      <div
        className="block"
      >
        <div
          className="sub-title"
        >
          Attendance Percentage
        </div>
        <div
          className="data"
        >
          78.07
        </div>
      </div>
    </div>
    <div>
      <div
        className="block"
      >
        <div
          className="sub-title"
        >
          Guidance Counselor
        </div>
        <div
          className="data"
        >
          Kent, Clark
        </div>
      </div>
      <div
        className="block"
      >
        <div
          className="sub-title"
        >
          Guidance Counselor Email
        </div>
        <div
          className="data"
        >
          kent.clark@newvisionshs.org
        </div>
      </div>
      <div
        className="block"
      >
        <div
          className="sub-title"
        >
          Home Phone Number
        </div>
        <div
          className="data"
        >
          917-590-3079
        </div>
      </div>
      <div
        className="block"
      >
        <div
          className="sub-title"
        >
          Advisor
        </div>
        <div
          className="data"
        >
          Kyle, Selina
        </div>
      </div>
      <div
        className="block"
      >
        <div
          className="sub-title"
        >
          Official Class
        </div>
        <div
          className="data"
        >
          345
        </div>
      </div>
      <div
        className="block"
      >
        <div
          className="sub-title"
        >
          Total Credits Earned
        </div>
        <div
          className="data"
        >
          24
        </div>
      </div>
      <div
        className="block"
      >
        <div
          className="sub-title"
        >
          Has Passed Regents
        </div>
        <div
          className="data"
        >
          YES
        </div>
      </div>
      <div
        className="block"
      >
        <div
          className="sub-title"
        >
          Status
        </div>
        <div
          className="data"
        >
          Active
        </div>
      </div>
      <div
        className="block"
      >
        <div
          className="sub-title"
        >
          Admin Date
        </div>
        <div
          className="data"
        >
          Sept 2016
        </div>
      </div>
    </div>
  </div>
</div>
`;
